packages/kbn-search-index-documents/components/result/rich_result_header.tsx

```diff
@@ -118,22 +120,39 @@ const MetadataPopover: React.FC<MetaDataProps> = ({
       </EuiFlexGroup>
       {onDocumentDelete && (
         <EuiPopoverFooter>
-          <EuiButton
-            iconType="trash"
-            color="danger"
-            size="s"
-            data-test-subj="deleteDocumentButton"
-            onClick={(e: React.MouseEvent<HTMLElement>) => {
-              e.stopPropagation();
-              onDocumentDelete();
-              closePopover();
-            }}
-            fullWidth
+          <EuiToolTip
+            content={
+              /* for serverless search users hasDeleteDocumentsPrivilege flag indicates if user has privilege to delete documents, for stack hasDeleteDocumentsPrivilege would be undefined */
+              hasDeleteDocumentsPrivilege !== undefined && !hasDeleteDocumentsPrivilege
+                ? i18n.translate(
+                    'searchIndexDocuments.result.header.metadata.deleteDocumentToolTip',
+                    {
+                      defaultMessage: 'You do not have permision to delete documents',
+                    }
+                  )
+                : undefined
+            }
+            position="bottom"
+            data-test-subj="deleteDocumentButtonToolTip"
           >
-            {i18n.translate('searchIndexDocuments.result.header.metadata.deleteDocument', {
-              defaultMessage: 'Delete document',
-            })}
-          </EuiButton>
+            <EuiButton
+              iconType="trash"
+              color="danger"
+              size="s"
+              isDisabled={hasDeleteDocumentsPrivilege !== undefined && !hasDeleteDocumentsPrivilege}
+              data-test-subj="deleteDocumentButton"
+              onClick={(e: React.MouseEvent<HTMLElement>) => {
+                e.stopPropagation();
+                onDocumentDelete();
+                closePopover();
+              }}
+              fullWidth
+            >
+              {i18n.translate('searchIndexDocuments.result.header.metadata.deleteDocument', {
```

Comment:
nitpick: FormattedMessage instead of i18n.translate

Reply:
Curious why FormattedMessage instead of i18n.translate?

Reply:
It was hard to find :)  long time ago I found this Best Practices from `i18n` team. Where is said: "You should use <FormattedMessage> most of the time.". 
[link](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/GUIDELINE.md#best-practices)

Reply:
Thank you for sharing this Yan!